lua/lualine/components/filetype.lua

```diff
@@ -31,9 +31,12 @@ function M:apply_icon()
   local icon, icon_highlight_group
   local ok, devicons = pcall(require, 'nvim-web-devicons')
   if ok then
-    local f_name, f_extension = vim.fn.expand('%:t'), vim.fn.expand('%:e')
-    f_extension = f_extension ~= '' and f_extension or vim.bo.filetype
-    icon, icon_highlight_group = devicons.get_icon(f_name, f_extension)
```

Comment:
Do we need to remove these ? Can't we use get_icon_by_filetype when get_icon returns nil as fallback?

Reply:
This might work ... playing around with `get_icon` this might be a valid simplifcation:
```lua
icon, icon_highlight_group = devicons.get_icon(vim.fn.expand('%:t'))
if icon == nil then
  icon, icon_highlight_group = devicons.get_icon_by_filetype(vim.bo.filetype)
end
```
as `get_icon` determines the extension from the name if no extension (2nd argument) is given.

Reply:
That looks cleaner . You can change it to this.

I was thinking about keeping the older filetype based check because the webdeviconcs pr stated the list added wasn't exhaustive . But now looking at it I think in the cases where older filetype check would matter ar already supported by `get_icon_by_filetype` . So we can remove that.